### ISSUE TITLE
Allow Collapse of levels in Code Browser

### ISSUE THREAD

**Issue by cgchoffman** [Komodo/KomodoEdit#2722]
### Short Summary

I can't seem to collapse anything in the new code browser and I would definitely like to.
I just want to see the function names in the screenshot but can't hide their children.  This would get much more noticeable with a larger file:
![capture](https://user-images.githubusercontent.com/655154/28692777-e77ceab0-72d6-11e7-917e-ad233c69280c.PNG)

**Comment by Naatan** [comment]
It's not supposed to show STRU symbols in there, I'd imagine once that is fixed you shouldn't have a need for collapsing. I'd like to avoid collapsing structures as I always found it to be very annoying in past versions of Komodo. And frankly once the structure is fixed up (STRU isnt showing) I don't think you'd have much of a need for it.

**Comment by cgchoffman** [comment]
> I'd like to avoid collapsing structures as I always found it to be very annoying

Annoying how?  I always use it.  Removing `STRU` will not fix the issue for me.  It will make the list shorter but it will still list way too much information.  You don't plan to have code folding at all when this is done?  Is that a researched decision or personal choice?  I think that is a major mistake, personally.

**Comment by Defman21** [comment]
I agree with Carey on this one. You can make it a pref or something, but some users will need the collapsing feature. (e.g. they are reviewing function definitions and they don't want to see their vars and everything else). More over, you can sync the tree with the editor so if I collapse a function, the code browser will collapse it too :)

**Comment by Naatan** [comment]
It's just a personal opinion at this stage, I'm not sold on it but that's the way I'm weighing at the moment. I don't have an issue with supporting (not defaulting to) folding other than the time requirement.
